future/atomicTransactionComposer.go

```diff
@@ -0,0 +1,681 @@
+		if err != nil {
+			return nil, nil, err
+		}
+
+		stxs[i] = stxBytes
+		txids[i] = txid
+	}
+
+	return stxs, txids, nil
+}
+
+func (txSigner LogicSigAccountTransactionSigner) Equals(other TransactionSigner) bool {
+	if castedSigner, ok := other.(LogicSigAccountTransactionSigner); ok {
+		otherJson, err := json.Marshal(castedSigner)
+		if err != nil {
+			return false
+		}
+
+		selfJson, err := json.Marshal(txSigner)
+		if err != nil {
+			return false
+		}
+
+		return string(otherJson) == string(selfJson)
+	}
+	return false
+}
+
+/**
+ * TransactionSigner that can sign transactions for the provided MultiSig Account
+ */
+type MultiSigAccountTransactionSigner struct {
+	Msig crypto.MultisigAccount
+	Sks  [][]byte
+}
+
+func (txSigner MultiSigAccountTransactionSigner) SignTransactions(txGroup []types.Transaction, indexesToSign []int) ([][]byte, []string, error) {
+	stxs := make([][]byte, len(indexesToSign))
+	txids := make([]string, len(indexesToSign))
+	for i, pos := range indexesToSign {
+		var unmergedStxs [][]byte
+		var txid string
+		for _, sk := range txSigner.Sks {
+			tempTxid, unmergedStxBytes, err := crypto.SignMultisigTransaction(sk, txSigner.Msig, txGroup[pos])
+			txid = tempTxid
+			if err != nil {
+				return nil, nil, err
+			}
+
+			unmergedStxs = append(unmergedStxs, unmergedStxBytes)
+		}
+
+		if len(txSigner.Sks) > 1 {
+			tempTxid, stxBytes, err := crypto.MergeMultisigTransactions(unmergedStxs...)
+			txid = tempTxid
+			if err != nil {
+				return nil, nil, err
+			}
+
+			stxs[i] = stxBytes
+			txids[i] = txid
+		} else {
+			stxs[i] = unmergedStxs[0]
+			txids[i] = txid
+		}
+	}
+
+	return stxs, txids, nil
+}
+
+func (txSigner MultiSigAccountTransactionSigner) Equals(other TransactionSigner) bool {
+	if castedSigner, ok := other.(MultiSigAccountTransactionSigner); ok {
+		otherJson, err := json.Marshal(castedSigner)
+		if err != nil {
+			return false
+		}
+
+		selfJson, err := json.Marshal(txSigner)
+		if err != nil {
+			return false
+		}
+
+		return string(otherJson) == string(selfJson)
+	}
+	return false
+}
+
+var ABI_RETURN_HASH = []byte{0x15, 0x1f, 0x7c, 0x75}
+
+/** Represents an unsigned transactions and a signer that can authorize that transaction. */
+type TransactionWithSigner struct {
+	/** An unsigned transaction */
+	Txn types.Transaction
+	/** A transaction signer that can authorize txn */
+	Signer TransactionSigner
+}
+
+/** Represents the output from a successful ABI method call. */
+type ABIResult struct {
+	/** The TxID of the transaction that invoked the ABI method call. */
+	TxID string
+	/**
+	 * The raw bytes of the return value from the ABI method call. This will be empty if the method
+	 * does not return a value (return type "void").
+	 */
+	RawReturnValue []byte
+	/**
+	 * The return value from the ABI method call. This will be undefined if the method does not return
+	 * a value (return type "void"), or if the SDK was unable to decode the returned value.
+	 */
+	ReturnValue abi.Value
```

Comment:
This could be just `interface{}`, I don't see the advantage to using `abi.Value`